enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -0,0 +1,222 @@
+
+This proposal is not relevant to MicroShift, as it does not run with multiple users.
+
+### Implementation Details/Notes/Constraints
+
+#### cluster-kube-apiserver-operator
+
+The cluster-kube-apiserver-operator relies on the authentication configuration in the following cases:
+- the `WebhookTokenAuthenticator` observer observes the `webhookTokenAuthenticator` field of the Authentication CR and if `kubeConfig` secret reference is set it uses the contents of this secret as a webhook token authenticator for the API server; it also takes care of synchronizing this secret to the `openshift-kube-apiserver` namespace
+- the `AuthMetadata` observer sets the `OauthMetadataFile` field of the CR with the path for a ConfigMap referenced by the authentication config
+
+The operator must watch the Authentication CR for changes, and when it detects an external OIDC provider configuration, it must make the following changes, in order to update its configuration to use the external provider:
+- the `WebhookTokenAuthenticator` observer must replace the secret used as a webhook token authenticator for the API server with the one of the new provider, and synchronize it to the `openshift-kube-apiserver` namespace
+- the `AuthMetadata` observer must be stopped, as it is not relevant in the case of the external OIDC provider
+
+Note that the operator must first validate the new provider configuration before proceeding with the deactivation of the built-in OAuth stack. Also, in case the authentication configuration gets changed back to the built-in OAuth server, the operator must revert these changes and bring the kube-apiserver and relevant resources back to the original state of affairs.
+
+#### cluster-authentication-operator
+
+When the built-in OAuth server is used for authentication (the default and original cluster state), the cluster-authentication-operator manages all controllers and resources related to it; notably, it manages the deployments of the oauth-server and oauth-apiserver, and manages resources such as the respective namespaces, service accounts, role bindings, services, OAuth clients, etc. Moreover, it monitors the status of its operands, making sure that the routes/services are healthy and reachable, and updates its operator status accordingly.
+
+In case an external OIDC provider is configured for authentication, then these controllers and resources are not useful or irrelevant. Therefore, the operator must watch the Authentication CR, and when it detects a valid external OIDC provider configuration, it must stop its controllers and remove its resources. The operator will not be monitoring the oauth-server and oauth-apiserver any longer, however it must monitor the external OIDC provider for reachability and health, and advertise the result in its status; it must either adapt the functionality of its monitoring controllers or use a new controller for that.
```

Comment:
you probably don't want the controllers to be actually stopped but rather turn them into a state where they remove the state they otherwise push to the cluster.